src/extension.ts

```diff
@@ -145,6 +158,106 @@ async function copyConnectionString(node: IAzureNode<MongoAccountTreeItem | DocD
+	if (!nodes) {
+		nodes = await askForDocuments();
+	}
+	const documents = await parseDocumentsForErrors(nodes);
+
+	if (!collectionNode) {
+		collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+	}
+	let result: string;
+	if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+		const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+		//tslint:disable:no-non-null-assertion
+		result = await collectionTreeItem!.executeCommand('insertMany', [JSON.stringify(documents)]);
+	} else {
+		result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, nodes);
+	}
+	await collectionNode.refresh();
+	await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+	let files: vscode.Uri[] = await vscode.workspace.findFiles("*.json");
+	let jsonDocuments: (vscode.QuickPickItem & { uri: vscode.Uri })[] = [];
+	let items: (vscode.QuickPickItem & { uri: vscode.Uri })[] = files.map(file => {
+		return { uri: file, label: vscode.workspace.asRelativePath(file) };
+	});
+	let pickAgain: string = "Pick again";
+	let discontinue = "Discontiue import";
+	while (!jsonDocuments.length) {
+		jsonDocuments = await ext.ui.showQuickPick(items, { canPickMany: true, placeHolder: "Choose a document to upload. Hit Escape to Cancel" });
+		if (!jsonDocuments.length) {
+			let action: string = await vscode.window.showWarningMessage("No document picked. Want to pick again?", pickAgain, discontinue);
+			if (action === discontinue) {
+				throw new UserCancelledError();
+			}
+		}
+	}
+	return jsonDocuments.map(choice => choice.uri);
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+	const parseResult = await parseDocuments(nodes);
+	const documents = parseResult[0];
+	const errors: string[] = parseResult[1];
+	if (errors.length > 0) {
+		ext.outputChannel.show();
+		throw new Error(`Errors found in the following documents: ${errors.join(',')}.\nPlease fix these and try again.`);
+	}
+	return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(nodes: vscode.Uri[]): Promise<[any[], string[]]> {
+	let documents = [];
+	let errors = {};
+	for (let node of nodes) {
+		const document = (await vscode.workspace.openTextDocument(node));
+		const text = document.getText();
+		let parsed;
+		try {
+			parsed = JSON.parse(text);
+		} catch (e) {
+			const err = parseError(e);
+			const fileName = node.path.split('/').pop();
+			errors[fileName] = err;
+			ext.outputChannel.appendLine(`${fileName}:\n${err}`);
+			await vscode.window.showTextDocument(document);
```

Comment:
What's the thoughts for showing all these files? This could get unwieldy if I'm uploading a bunch of files and several of them have errors. For example, the VS Code explorer lets me select any file type and anything that's not 'json' will fail:
![screen shot 2018-09-27 at 11 02 12 am 2](https://user-images.githubusercontent.com/11282622/46165238-ffcd2800-c244-11e8-92c5-e2f29023900a.png)

Reply:
Nice catch. I remember testing the multiselect behavior, and it looked like it would not show unless all the files were .json. 
I tested it some more just now, and it looks like the group of files does not determine context menu as much as _which file_ was right-clicked. I right-clicked on the .cs file for the screenshot.
![image](https://user-images.githubusercontent.com/30420973/46167424-9b14cc00-c24a-11e8-95e6-39895c83fe06.png)

So my code checks for the clicked file, but made an assumption that all files should support the context item => all are jsons.

Reply:
I've filed an issue for the symptom you mentioned (selecting a non JSON file) [here](https://github.com/Microsoft/vscode/issues/59782). I'm handling this behavior in the next push. 

I understand where the issue with too many errors. It looks like vscode would overwrite the previously opened document (it is temporary, so the tab-title is italicized). I might scrap opening the documents  entirely, let me play with that.